edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/DeviceScopeController.cs

```diff
@@ -139,19 +164,16 @@ async Task<EdgeHubScopeResult> HandleGetDeviceAndModuleOnBehalfOfAsync(string ac
 
             IEdgeHub edgeHub = await this.edgeHubGetter;
 
-            // Check if the actor has provided the originating EdgeHub ID,
-            // if not, then we must be by definition by the origin.
-            string originatingEdge = edgeHub.GetEdgeDeviceId();
-            if (this.Request.Headers.TryGetValue(Constants.OriginEdgeHeaderKey, out StringValues originHeader) && originHeader.Count > 0)
+            if (!AuthChainHelpers.TryGetTargetDeviceId(request.AuthChain, out string originatingEdgeDevice))
             {
-                originatingEdge = originHeader.First();
+                originatingEdgeDevice = actorDeviceId;
             }
 
             // We must always forward the call further upstream first,
             // as this is invoked for refreshing an identity on-demand,
             // and we don't know whether our cache is out-of-date.
             IDeviceScopeIdentitiesCache identitiesCache = edgeHub.GetDeviceScopeIdentitiesCache();
-            await identitiesCache.RefreshServiceIdentityOnBehalfOf(targetId, originatingEdge);
+            await identitiesCache.RefreshServiceIdentityOnBehalfOf(targetId, originatingEdgeDevice);
```

Comment:
>originatingEdgeDevice [](start = 77, length = 21)

I think this still needs a default (current deviceId) for the bottom level case

Reply:
Emm, I am not sure I follow.. what should be set to the default?

Reply:
RefreshServiceIdentityOnBehalfOf() will throw if the OnBehalfOfDevice is empty, which can happen here if the caller supplies an empty authchain.  We can either param check for the empty case and return a meaningful error, or default the OnBehalfOfDevice to the current Edge device ID like the previous code did.


---
In reply to: [605145686](https://github.com/Azure/iotedge/pull/4735#discussion_r605145686) [](ancestors = 605145686)

Reply:
yes, and I default to actorDeviceId if the authchain is empty - https://github.com/Azure/iotedge/pull/4735/files#diff-b372abe320493b2949d978c985c8e09cf07491f23268ac948f81ac3bc344a8b8R154